src/Microsoft.Azure.WebJobs.Host/Queues/Listeners/QueueListener.cs

```diff
@@ -271,6 +271,7 @@ private TaskSeriesCommandResult CreateSucceededResult()
             {
                 if (!await _queueProcessor.BeginProcessingMessageAsync(message.SdkObject, cancellationToken))
                 {
+                    await _queueProcessor.MoveMessageToPoisonQueueAsync(message.SdkObject, cancellationToken);
```

Comment:
Customers can override BeginProcessingMessage and return false for their own reasons. I don't think it's safe to assume that if they return false the message should be moved to poison queue.

Instead, move this into BeginProcessingMessage.